File: dotnet/architecture-patterns/src/Bad/OrderControllerBad.cs
// BAD EXAMPLE — INTENTIONALLY NON-COMPLIANT
// Excluded from compilation via the test project's direct-compile approach.
//
// Violates the layered architecture rules enforced by the fitness function.
//
// Expected Semgrep violations:
//   Line 5  — no-repository-using-in-controller (using Repositories in controller file)
//   Line 18 — no-repository-using-in-controller (new OrderRepository() in controller)
//
// NetArchTest violations (detected in ArchitectureTests.cs):
//   - Controllers namespace depends on Repositories namespace

using ArchitecturePatterns.Repositories; // VIOLATION: controller imports repository namespace

namespace ArchitecturePatterns.Bad;

public class OrderControllerBad
{
    // VIOLATION: controller holds direct reference to repository
    private readonly OrderRepository _repository = new OrderRepository();

    public Dictionary<string, object> GetOrder(string orderId)
    {
        // VIOLATION: data access logic in the controller layer
        var order = _repository.FindById(orderId);
        if (order is null)
            return new Dictionary<string, object> { ["error"] = "not_found" };

        return new Dictionary<string, object>
        {
            ["orderId"] = order.OrderId,
            ["status"]  = order.Status,
        };
    }
}

File: Node/architecture-patterns/src/bad/order-controller-bad.js
'use strict';
/**
 * src/bad/order-controller-bad.js — INTENTIONALLY NON-COMPLIANT
 *
 * Violates the layered architecture rules enforced by the fitness function.
 * Run dependency-cruiser to see all violations:
 *
 *   npx depcruise --config .dependency-cruiser.cjs src/bad
 *
 * Expected violations:
 *   Line 14 — no-repository-in-controller (controller imports repository directly)
 *
 * Semgrep will also flag:
 *   Line 14 — no-repository-import-in-controller
 */

// VIOLATION: controller imports repository directly — bypasses service layer
const repository = require('../repositories/order-repository');

function getOrder(req, res) {
  // VIOLATION: business logic and data access in the controller
  const order = repository.findById(req.params.orderId);
  if (!order) {
    return res.status(404).json({ error: 'not_found' });
  }
  return res.json(order);
}

module.exports = { getOrder };
